thing.go

```diff
@@ -4,57 +4,73 @@ import (
 	"time"
 )
 
-// Thing is the top level datastructure representing an individual thing as
+// Thing is the top level data structure representing an individual thing as
 // indexed on some upstream data provider.
 type Thing struct {
-	// Title is a short descriptive lable for the thing
+	// Title is a short descriptive label for the thing
 	Title string `json:"title"`
 
 	// Description is a long form description about the thing
 	Description string `json:"description"`
 
-	// Metadata is a comma separated tags - additional metadata about the Thing,
-	// that isn't necessarily encoded into the feed.
-	Metadata string `json:"-"`
-
-	// Category is a thingfulx.Category instance - choose from the declared categories
-	Category *Category `json:"-"`
+	// IndexedAt is a Time object indicating the timestamp when the thing was
+	// indexed by Thingful
+	IndexedAt time.Time `json:"indexed_at"`
 
 	// Webpage is a URL to a human intelligible page about the thing. This
 	// doesn't have to be a unique page for the thing as typically one might not
 	// be available. In that case, this could be a link to a page describing the
 	// type of thing, or just a link to the main homepage of the project.
-	Webpage string `json:"webpage"`
+	Webpage string `json:"-"`
 
-	// DataURL is a string containing the actual URL we go to fetch data from for
-	// this thing. This string must be unique for every thing indexed. We can
-	// achieve this for things that don't make a real unique URL available by
-	// adding a synthetic identifier to the URL string, i.e.
-	// http://example.com?city=Indiana#section_id=1234
-	DataURL string `json:"datasource"`
-
-	// IndexedAt is a Time object indicating the timestamp when the thing was
-	// indexed by Thingful
-	IndexedAt time.Time `json:"indexed_at"`
+	// DataURL is an embedded data structure which describes the actual data entry point.
+	Endpoint Endpoint `json:"endpoint"`
 
-	// Lng is the longitude of the Thing, represented as a decimal value
-	Lng float64 `json:"longitude"`
+	// Metadata is a data structure containing additional metadata about the Thing.
+	Metadata []Metadata `json:"metadata"`
 
-	// Lat is the latitude of the Thing, represented as a decimal value
-	Lat float64 `json:"latitude"`
+	// Location is an embedded data structure which describes location data.
+	Location Location `json:"location"`
 
-	// Provider is a data structure containing information about the upstream
-	// data provider.
-	Provider *Provider `json:"-"`
+	// Provider is an embedded data structure which describes the data provider.
+	Provider Provider `json:"provider"`
 
 	// Visibility indicates what level of access the upstream data provider
 	// offers, Closed, Shared or Open.
 	Visibility Visibility `json:"visibility"`
```

Comment:
`Visibility` and `Authentication` are slightly different I think. The concept of visibility was taken from the ODI definition of Open, Shared and Closed data (https://theodi.org/blog/closed-shared-open-data-whats-in-a-name). My understanding is that the ODI definition involves notions of how data is licensed and its 'type' (government vs commercial vs personal).

If we - like Moeen - don't care about ODI we can either remove it or treat it as metadata. I would go for moving it into the metadata section so it can potentially be used with semantic search.